tests/vanilla/utils/atomWithReset.test.ts

```diff
@@ -0,0 +1,56 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { createStore } from 'jotai/vanilla'
+import { RESET, atomWithReset } from 'jotai/vanilla/utils'
+
+describe('atomWithReset', () => {
+  let initialValue: number
+  let testAtom: any
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+    initialValue = 10
+    testAtom = atomWithReset(initialValue)
+  })
+
+  it('should create an atom with initial value', () => {
+    const { init } = testAtom
+    expect(init).toBe(initialValue)
+  })
```

Comment:
This is testing the internal behavior. While it's correct, let's remove from the atomWithReset test.

Reply:
OK